src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -176,6 +179,10 @@ export class Kernel implements IKernel {
                         }
                     } catch (ex) {
                         traceError('failed to create INotebook in kernel', ex);
+                        sendKernelTelemetryEvent(options.document.uri, Telemetry.NotebookStart, stopWatch.elapsedTime, {
+                            failed: 'true',
+                            reason: getKernelFailureReason(ex)
```

Comment:
Regardless of Remote, local, zmq/jupyter, if somethign fails, we know why it failed & we capture the reason here (look at `getkernelFailureReason`).

I.e. for Native Notebooks, this is the choke point for error handling/logging.

Hopefully I can find a similar point in WebViews & Interactive Window to handle all errors in one place & capure the logs in one place.